### ISSUE TITLE
Feature Request: Gradient checkpointing to fit larger models.

### ISSUE THREAD

**Issue by dimeldo** [bme-chatbots/dialogue-generation#12]
Gradient checkpointing is implemented here: https://github.com/nshepperd/gpt-2
Maybe we can add this somehow so we can train larger models like gpt2-large?

**Comment by Mrpatekful** [comment]
Currently we are using huggingface/pytorch_transformers for the model implementations but in the near future (2-3 weeks) we will release our own transformers repository, which is based on NVIDIA/MegatronLM, where checkpointing is implemented.

**Comment by Mrpatekful** [comment]
Gradient checkpointing has been added and is available with `--checkpointed` flag.
